compiler/rustc_mir_build/src/thir/pattern/_match.rs

```diff
@@ -820,7 +820,11 @@ enum Constructor<'tcx> {
     Single,
     /// Enum variants.
     Variant(DefId),
-    /// String literals
+    /// String literals. These have their own variant, because they behave essentially like
+    /// `&[u8]`, but there's no way to pattern match on parts of strings. So we could convert them
+    /// to slice constructors, but then we'd have to special case other code sites to not treat them
+    /// *exactly* like slice constructors. It seems simpler overall to just give them their own
+    /// variant.
```

Comment:
Is it true that `Constructor` is used only for exhaustiveness checking, but does not actually effect MIR construction for pattern matching?

If yes, I think it would be good to mention that in the `enum Constructor` doc comment (but doesn't have to be in this PR).

Reply:
I don't actually know. Decisions on MIR match construction may be made on information obtained via `Constructor`.

Reply:
Yes, it's only for exhaustiveness

Reply:
I know because I've changed Constructor without any code breaking outside of this `_match.rs`.